docs/en_us/platform_api/source/courses/blocks.rst

```diff
@@ -0,0 +1,323 @@
+&all_blocks=true&requested_fields=graded,format,student_view_multi_device
+
+=====================
+Query Parameters
+=====================
+
+* all_blocks: (boolean) Provide a value of ``true`` to return all blocks.
+
+* block_counts: (list) Specify the types of blocks for which to return a count
+  of that type (including child blocks). Example: ``block_counts=video,problem``.
+
+* block_types_filter: (list) Specify the types of blocks to be included in results.
+  Possible values are: ``sequential``, ``vertical``, ``html``, ``problem``,
+  ``video`` and ``discussion``. Example: ``block_types_filter=problem,html``.
+
+* course_id: (string, required) The URL-encoded ID of the course whose block
+  data you are requesting. Example: ``course_id=edX%2FDemoX%2FDemo_Course``.
+
+* depth: (integer or ``all``) Specify how far in the course blocks hierarchy
+  to traverse down. A value of ``all`` specifies the entire hierarchy. The
+  default value is ``0``. Example: ``depth=all``.
+
+* requested_fields: (list) Specify the fields to return for each block, in
+  addition to ID, type, and display_name, which are always returned. For the
+  list of possible fields, see the fields listed under ``blocks`` in the
+  **Response Values** section below. Example:
+  ``requested_fields=graded,format,student_view_multi_device``.
+
+* return_type: (string) Specify the data type in which the block data is
+  returned. Supported values are ``dict`` and ``list``. The default value is
+  ``dict``.
+
+* student_view_data: (list) Specify for which block types the JSON data is
+  returned in the "student_view_data" response value. Example:
+  ``student_view_data=video``.
+
+* username: (string) Required, unless ``all_blocks`` is specified. Example:
+  ``username=anjali``.
+
+
+=====================
+Response Values
+=====================
+
+The following fields are returned with a successful response.
+
+* root: The ID of the root node of the requested course block structure.
+
+* blocks: A dictionary that maps block usage IDs to a collection of information
+  about each block. Each block contains the following fields.
+
+  * id: (string) The usage ID of the block.
+
+  * type: (string) The type of block. Possible values include ``course``,
+    ``chapter``, ``sequential``, ``vertical``, ``html``, ``problem``,
+    ``video``, and ``discussion``. The type can also be the name of a custom
+    type of block.
+
+  * display_name: (string) The display name of the block.
+
+  * children: (list) If the block has child blocks, a list of IDs of the child
+    blocks. Returned only if ``children`` is included in the ``requested_fields``
+    query parameter.
+
+  * block_counts: (dict) For each block type specified in the ``block_counts``
+    parameter to the endpoint, the aggregate number of blocks of that type for
+    this block and all of its descendants. Returned only if the ``block_counts``
+    query parameter contains this block's type.
```

Comment:
Description of block_counts in blocks/views.py should also be updated to correct this.